### PR TITLE
Implement break statement

### LINKED ISSUE #22: Break statement is not implemented in compiler

### PR DESCRIPTION
**Pull request by oczekrb**
Closes #22

### CODE REVIEW

**Comment by jarkonik** [comment]
I we can use something like this everywhere we iterate over statements to skip everything that comes after break
`
fn walk_statements(&mut self, stmts: Vec<Statement>) -> Value {
        let mut last_stmt = Value::Null;
        for stmt in stmts {
            last_stmt = self.walk(&stmt);
            if matches!(last_stmt, Value::Break) {
                break;
            }
        }
       last_stmt
}
`

File: src/compiler/mod.rs [hunk 1]
```diff
@@ -184,11 +186,20 @@ impl Visitor<Value> for Compiler {
             Value::Bool(b) => {
                 self.builder.build_cond_br(&b, &then_block, &else_block);
                 self.builder.position_builder_at_end(&then_block);
+
+                let mut is_break = false;
                 for stmt in &expr.body {
                     self.walk(stmt);
+
+                    if matches!(stmt, expression::Expression::Break) {
```
Comment (jarkonik) [inline src/compiler/mod.rs:194]: couldn't we just have?
```
if matches!(stmt, expression::Expression::Break) {
  self.builder.create_br(&after_if_block);
  brak;
}
```

File: src/compiler/mod.rs [hunk 2]
```diff
@@ -696,6 +725,7 @@ impl Compiler {
                 Value::Null => unreachable!(),
                 Value::String(_) => todo!(),
                 Value::Function { .. } => (),
+                Value::Break => todo!(),
```
Comment (jarkonik) [inline src/compiler/mod.rs:728]: there is another exception macro that would fix more nicely here - `unreachable!()`

File: src/compiler/var.rs [hunk 3]
```diff
@@ -14,6 +14,7 @@ pub enum Var {
         typ: llvm::Type,
         return_type: parser::Type,
     },
+    Break,
```
Comment (jarkonik) [inline src/compiler/var.rs:17]: Do we need to have `break` as a type of variable?

**Review by jarkonik — CHANGES_REQUESTED** [review]
Some questions/maybe change requests. Also clippy has crashed.

File: src/compiler/mod.rs (hunk 1 above)
Comment (oczekrb) [reply src/compiler/mod.rs:194]: not exactly because after loop block branch should be created only when in iteration will NOT meet break statement.

Comment (jarkonik) [reply src/compiler/mod.rs:194]: yeah, that is indeed the case in the second occurrence of this - you've added similar code 2 times in this PR. I think you are describing the other  occurrence, here the ``is_break``` variable seems to be only used once :)

**Review by jarkonik — APPROVED** [review]

**Pull request merged**